### ISSUE TITLE
[cmake] toplevel-parasite tool detection is backwards (Trac #1358)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1350]
The parasitic build system has no explicit tool detection phase. Instead, the parasitic version of `i3_project()` loads tool-related variables from the parent CMakeCache.txt based on the values passed to `USE_TOOLS`. Some projects with uncommon dependencies such as `astro`, however, wrap the body of their CMakeLists.txt in an `if (${TOOL}_FOUND)` clause, which will fail because the corresponding variable is not defined until explicitly requested.

Two solutions come to mind:
1. Eagerly load everything from the parent CMakeCache.txt, filtering out tool-related variables by looking for names that share a prefix with variables of the form `${TOOL}_FOUND`. This is icky but simple.
2. Replace the `if (${TOOL}_FOUND)` pattern with a function call, e.g. `if (tool_found(${TOOL}))`. This is more work and involves touching multiple projects, but is probably clearer in the long run.

Migrated from https://code.icecube.wisc.edu/ticket/1358
```json
{
    "status": "closed", 
    "changetime": "2019-01-12T00:06:17", 
    "description": "The parasitic build system has no explicit tool detection phase. Instead, the parasitic version of `i3_project()` loads tool-related variables from the parent CMakeCache.txt based on the values passed to `USE_TOOLS`. Some projects with uncommon dependencies such as `astro`, however, wrap the body of their CMakeLists.txt in an `if (${TOOL}_FOUND)` clause, which will fail because the corresponding variable is not defined until explicitly requested.\n\nTwo solutions come to mind:\n1. Eagerly load everything from the parent CMakeCache.txt, filtering out tool-related variables by looking for names that share a prefix with variables of the form `${TOOL}_FOUND`. This is icky but simple.\n2. Replace the `if (${TOOL}_FOUND)` pattern with a function call, e.g. `if (tool_found(${TOOL}))`. This is more work and involves touching multiple projects, but is probably clearer in the long run.", 
    "reporter": "jvansanten", 
    "cc": "nega, david.schultz", 
    "resolution": "fixed", 
    "_ts": "1547251577858784", 
    "component": "cmake", 
    "summary": "[cmake] toplevel-parasite tool detection is backwards", 
    "priority": "major", 
    "keywords": "", 
    "time": "2015-09-21T12:36:01", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20151111T17:00:58: 
* david.schultz changed cc from "" to "nega, david.schultz"
* david.schultz changed milestone from "" to "January 2016"
* david.schultz changed priority from "normal" to "major"

**Comment by icecube-trac** [comment]
Trac update at 20151214T21:12:56: 
* jvansanten commented:

> In changeset 2374:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2374"
> Greedily load tool configuration in parasitic build, using a tool list shared
> with the main build system. Along the way, ensure that important tool variables
> actually appear in the cache. Fixes #1350
> ```


* jvansanten changed status from "new" to "closed"
* jvansanten changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20160318T21:14:09: 
* olivas changed milestone from "January 2016" to ""
* olivas commented:

> Milestone January 2016 deleted

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:06:17: nwhitehorn commented:

> In changeset 170161:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170161"
> Greedily load tool configuration in parasitic build, using a tool list shared
> with the main build system. Along the way, ensure that important tool variables
> actually appear in the cache. Fixes #1350
> ```
